pkg/reconciler/autoscaling/kpa/scaler.go

```diff
@@ -116,7 +119,14 @@ func newScaler(ctx context.Context, psInformerFactory duck.InformerFactory, enqu
 func paToProbeTarget(pa *pav1alpha1.PodAutoscaler) string {
 	svc := pkgnet.GetServiceHostname(pa.Status.ServiceName, pa.Namespace)
 	port := networking.ServicePort(pa.Spec.ProtocolType)
-	return fmt.Sprintf("http://%s:%d/", svc, port)
+
+	httpDest := url.URL{
+		Scheme: "http",
+		Host:   fmt.Sprintf("%s:%d", svc, port),
```

Comment:
```suggestion
		Host:   svc + ":" + strconv.Itoa(port),
```

Reply:
I'd also recommend doing
```
u, _ := url.Parse(fmt.Sprintf("http://%s:%d/", svc, port))
u.Path = path.Join(u.Path, probePath)
return u.String()
```
perhaps with error handling, but we kind of know it won't fail here...

Reply:
But why "u.Path = path.Join(u.Path, probePath)" since we already know u.Path should be empty by definition in above line

Reply:
Is it empty? Or is it "/"? Your path does it have "/" prefix or not. This takes care of all the cases.

Reply:
hmm, makes sense